agent-control/src/sub_agent/on_host/supervisor.rs

```diff
@@ -319,19 +335,28 @@ impl NotStartedSupervisorOnHost {
 
 /// From the `ExitStatus`, send appropriate event and emit logs, return exit code.
 fn handle_termination(
+    exec_id: &str,
     exit_status: ExitStatus,
-    internal_event_publisher: &EventPublisher<SubAgentInternalEvent>,
+    health_publisher: EventPublisher<(String, HealthWithStartTime)>,
     agent_id: &AgentID,
-    bin: String,
+    bin: &str,
     start_time: SystemTime,
 ) -> i32 {
     if !exit_status.success() {
         let unhealthy: Unhealthy = Unhealthy::new(exit_status.to_string()).with_status(format!(
             "process exited with code: {:?}",
             exit_status.code().unwrap_or_default()
         ));
-        internal_event_publisher
-            .publish_health_event(HealthWithStartTime::new(unhealthy.into(), start_time));
+
+        let _ = health_publisher
+            .publish((
+                exec_id.to_string(),
+                HealthWithStartTime::new(unhealthy.into(), start_time),
+            ))
+            .map_err(|err| {
+                error!("Error saving health status for {exec_id}: {err}");
+            });
```

Comment:
Nit: An `if let Err` might make more sense in this case, where we are not transforming the error.